doc/index.md

```diff
@@ -1032,68 +1050,153 @@ On success, returns the response [*headers*](#http.headers) and a [*stream*](#st
+
 
 ### `server:events()` <!-- --> {#http.server:events}
 
+Returns a string or nil. The system searches for `'r'` for `POLLIN` or `'w'` for `POLLOUT`. Linux systems can also return `'p'` for `POLLPRI`. 
+
 
 ### `server:timeout()` <!-- --> {#http.server:timeout}
 
+Returns an integer or nil that represents a timeout in seconds.
+
 
 ### `server:empty()` <!-- --> {#http.server:empty}
 
+Returns true if the master socket and all client connection have been closed, false otherwise.
+
 
 ### `server:step()` <!-- --> {#http.server:step}
 
+Initiates a single response/request. This function can be used for fine control over the response processing. However, [*server:loop()*](#http.server:loop) is the recommended way to run the server for most applications. Returns nil, an error and an error message on failure. 
+
 
 ### `server:loop()` <!-- --> {#http.server:loop}
 
+Run the server as a blocking loop. The server will continue to listen and accept client requests until either server:pause() or server:close() is called, or an error is experienced. If this function takes advantange of cqueues stacking and when called from within a cqueue, the server will not block the main Lua application thread. 
+
+#### Example {#http.server:loop-example}
+
+```lua
+local http_server = server.listen {
+    host = 192.168.0.1;
+    port = 80;
+    onstream = function(server, stream)  ? end;
+}
+
+assert(http_server:loop())
+```
+
 
 ### `server:add_socket(socket)` <!-- --> {#http.server:add_socket}
 
+Add a new connection socket to the server for processing. The server will use the current `onstream` request handler and all `options` currently specified through the [*server.listen(options)*](#http.server.listen) constructor. `add_socket` can be used to process connection sockets obtained from an external source such as:
+  - Another cqueues thread with some other master socket.
+  - From inetd for start on demand daemons.
+  - A Unix socket with `SCM_RIGHTS`.
+
 
 ## http.socks
 
 Implements a subset of the SOCKS proxy protocol.
 
 ### `connect(uri)` <!-- --> {#http.socks.connect}
 
-  - `uri` is a string with the address of the SOCKS server. A scheme of `"socks5"` will resolve hosts locally, a scheme of `"socks5h"` will resolve hosts on the SOCKS server. If the URI has a userinfo component it will be sent to the SOCKS server as a username and password.
+`uri` is a string with the address of the SOCKS server. A scheme of `"socks5"` will resolve hosts locally, a scheme of `"socks5h"` will resolve hosts on the SOCKS server. If the URI has a userinfo component it will be sent to the SOCKS server as a username and password.
 
 Returns a *http.socks* object.
 
 
 ### `fdopen(socket)` <!-- --> {#http.socks.fdopen}
 
-  - `socket` should be a cqueues socket object
-
-Returns a *http.socks* object.
+Opens a file descrptor on `socket` and returns a *http.socks* object. `socket` should be a cqueues socket object
```

Comment:
> Opens a file descrptor on `socket`

I dont understand that.

`fdopen` takes an existing cqueues.socket to use as a basis for a http.socks object.